source/ota.c

```diff
@@ -3339,7 +3382,7 @@ OtaErr_t OTA_ActivateNewImage( void )
      * and not return unless there is a problem within the PAL layer. If it does return,
      * output an error message. The device may need to be reset manually.
      */
-    if( ( otaAgent.pOtaInterface != NULL ) && ( otaAgent.pOtaInterface->pal.activate != NULL ) )
+    if( ( otaAgent.state != OtaAgentStateStopped ) && ( otaAgent.pOtaInterface != NULL ) && ( otaAgent.pOtaInterface->pal.activate != NULL ) )
```

Comment:
I think having the check for `OtaAgentStateStopped` does not add much value as the shutdown happens concurrently in OTA agent thread.